stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/adapter/ChannelListItemAdapter.kt

```diff
@@ -1,19 +1,28 @@
 package io.getstream.chat.android.ui.channel.list.adapter
 
-import android.view.LayoutInflater
 import android.view.ViewGroup
-import io.getstream.chat.android.ui.R
+import io.getstream.chat.android.client.models.Channel
 import io.getstream.chat.android.ui.channel.list.adapter.diff.ChannelDiff
 import io.getstream.chat.android.ui.channel.list.adapter.viewholder.BaseChannelListItemViewHolder
-import io.getstream.chat.android.ui.channel.list.adapter.viewholder.BaseChannelViewHolderFactory
-import io.getstream.chat.android.ui.channel.list.adapter.viewholder.ChannelViewHolderFactory
+import io.getstream.chat.android.ui.channel.list.adapter.viewholder.ChannelListItemListenerContainer
+import io.getstream.chat.android.ui.channel.list.adapter.viewholder.ChannelListItemViewHolderFactory
 import io.getstream.chat.android.ui.utils.extensions.cast
 import io.getstream.chat.android.ui.utils.extensions.firstOrDefault
 
 public class ChannelListItemAdapter : BaseChannelListItemAdapter() {
 
-    public var viewHolderFactory: BaseChannelViewHolderFactory<BaseChannelListItemViewHolder> =
-        ChannelViewHolderFactory()
+    public var viewHolderFactory: ChannelListItemViewHolderFactory = ChannelListItemViewHolderFactory()
+
+    public var listenerContainer: ChannelListItemListenerContainer = ChannelListItemListenerContainer()
```

Comment:
The idea behind the original `ListenerContainer` is that the container itself will never change, only the listeners inside it, therefore it's safe for the ViewHolders to hold onto the container reference, and then fetch the current, up-to-date listener values from it when they need it.

This should probably be a `private val`, so that the container is never swapped for a new instance, because then already created ViewHolders would still reference the old instance.

Reply:
Oversight on my part; I'll change it.